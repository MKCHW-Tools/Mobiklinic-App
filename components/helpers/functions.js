import * as React from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {Alert} from 'react-native';
import axios from 'axios';
import {URLS} from '../constants/API';
import uniqWith from 'lodash/uniqWith';
import isEqual from 'lodash/isEqual';
import {
  LOGIN_USER,
  SIGN_UP_USER,
  REFRESH_TOKENS,
  DOWNLOAD_MUTATION,
  SAVE_LOCAL_USER_MUTATION,
} from '../graphql/Mutations';
import {RETRIEVE_LOCAL_USER_QUERY} from '../graphql/Queries';
import {ApolloClient, InMemoryCache} from '@apollo/client';

// Create an Apollo Client instance
const client = new ApolloClient({
  uri: 'https://staging.mobiklinic.com/graphql',
  cache: new InMemoryCache(),
});

// remove storage item
export const _removeStorageItem = async key => {
  try {
    await AsyncStorage.removeItem(key);
  } catch (error) {
    console.error('Failed to remove item from storage:', error);
  }
};

export const generateRandomCode = length => {
  const USABLE_CHARACTERS = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'.split('');

  return new Array(length)
    .fill(null)
    .map(
      () =>
        USABLE_CHARACTERS[Math.floor(Math.random() * USABLE_CHARACTERS.length)],
    )
    .join('');
};

export const MyDate = () => {
  const myDate = new Date();
  return `${
    myDate.getMonth() + 1
  }-${myDate.getDate()}-${myDate.getFullYear()} ${myDate.getHours()}:${myDate.getMinutes()}:${myDate.getSeconds()}`;
};

// tokens refresh
export const tokensRefresh = async user => {
  const refresh = user.tokens.refresh;

  try {
    const response = await client.query({
      query: REFRESH_TOKENS,
      variables: {
        refreshToken: refresh,
      },
      context: {
        headers: {
          Authorization: `Bearer ${refresh}`,
        },
      },
    });

    const {data} = response;
    const {tokensRefresh} = data;

    if (tokensRefresh === null) {
      await AsyncStorage.setItem(
        '@user',
        JSON.stringify({
          id: user.id,
          username: user.username,
          hash: null,
          tokens: user.tokens,
        }),
      );
      return null;
    }

    const {id, username, tokens, offline} = tokensRefresh;
    const {access, refresh: newRefresh} = tokens;

    await SAVE_LOCAL_USER({
      id,
      username,
      password: user.password,
      tokens: {
        access,
        refresh: newRefresh,
      },
      offline,
    });

    return {
      id,
      username,
      tokens: {
        access,
        refresh: newRefresh,
      },
      offline,
    };
  } catch (error) {
    console.error(error.message);
  }

  return null;
};

// retrrieve local user
export const RETRIEVE_LOCAL_USER = async () => {
  console.log('Retrieving local user');
  try {
    const response = await client.query({
      query: RETRIEVE_LOCAL_USER_QUERY,
    });

    const {data} = response;
    return data.retrieveLocalUser;
  } catch (error) {
    console.error(error.message);
    throw error;
  }
};

// save local user
export const SAVE_LOCAL_USER = async (user = {}) => {
  try {
    const HASH = cyrb53(user.password);

    const response = await client.mutate({
      mutation: SAVE_LOCAL_USER_MUTATION,
      variables: {
        input: {
          id: user.id,
          username: user.username,
          hash: HASH,
          tokens: user.tokens,
        },
      },
    });

    const {data} = response;
    return data.saveLocalUser;
  } catch (error) {
    console.error(error.message);
    throw error;
  }
};

export const cyrb53 = function (str, seed = 0) {
  let h1 = 0xdeadbeef ^ seed,
    h2 = 0x41c6ce57 ^ seed;

  for (let i = 0, ch; i < str.length; i++) {
    ch = str.charCodeAt(i);
    h1 = Math.imul(h1 ^ ch, 2654435761);
    h2 = Math.imul(h2 ^ ch, 1597334677);
  }

  h1 =
    Math.imul(h1 ^ (h1 >>> 16), 2246822507) ^
    Math.imul(h2 ^ (h2 >>> 13), 3266489909);
  h2 =
    Math.imul(h2 ^ (h2 >>> 16), 2246822507) ^
    Math.imul(h1 ^ (h1 >>> 13), 3266489909);
  return 4294967296 * (2097151 & h2) + (h1 >>> 0);
};

// download
export const DOWNLOAD = async data => {
  const {accessToken, items, userId, per_page} = data;

  try {
    const response = await client.query({
      query: DOWNLOAD_MUTATION,
      variables: {
        accessToken,
        items,
        per_page,
      },
      context: {
        headers: {
          Authorization: `Bearer ${accessToken}`,
        },
      },
    });

    const {data} = response;
    const {download} = data;

    return download;
  } catch (error) {
    console.error(error.message);
  }

  return false;
};

// signin a user
export const signIn = async data => {
  clearStorage();
  let {user, setIsLoading, setMyUser: setUser} = data;
  if (typeof user === undefined) {
    Alert.alert('Error', 'Provide your phone number and password');
    return;
  }

  const {username, password} = user;
  let hash = cyrb53(password);

  if (username === '' && password === '') {
    Alert.alert('Error', 'Provide your phone number and password');
    return;
  }

  let theUser = null;

  try {
    theUser = await RETRIEVE_LOCAL_USER();
  } catch (err) {
    console.log(err);
  }

  if (theUser !== null) {
    let myUser =
      theUser.username === username && theUser.hash === hash ? theUser : null;

    if (myUser) {
      setUser({
        id: myUser.id,
        username: myUser.username,
        tokens: myUser.tokens,
        offline: true,
      });
      setIsLoading(false);
      return;
    }
  }

  try {
    console.log('Starting network request');
    const response = await client.mutate({
      mutation: LOGIN_USER,
      variables: {
        username,
        password,
      },
    });

    const {data} = response;
    const {signIn} = data;

    const {id, tokens, offline} = signIn;

    await SAVE_LOCAL_USER({
      id,
      username,
      password,
      tokens: {
        access: tokens.access,
        refresh: tokens.refresh,
      },
    });

    const resources = ['ambulances', 'doctors', 'diagnosis'];

    if (
      DOWNLOAD({
        accessToken: tokens.access,
        items: resources,
        per_page: 10,
      })
    ) {
      setUser({
        id,
        username,
        tokens: {
          access: tokens.access,
          refresh: tokens.refresh,
        },
        offline,
      });
      setIsLoading(false);
    }
  } catch (error) {
    console.error(error);
    Alert.alert(
      'Failed to login',
      'Check your login details',
      [
        {
          text: 'Cancel',
          onPress: () => setIsLoading(false),
        },
      ],
      {
        cancelable: true,
        onDismiss: () => {
          setIsLoading(false);
        },
      },
    );
  }
};

// signup a user
export const signUp = async data => {
  const {
    firstName,
    lastName,
    phoneNumber,
    password,
    cPassword,
    eMail,
    setIsLoading,
    setProcess,
    setRegistered,
  } = data;

  setProcess('Registering, please wait!');

  if (
    firstName === '' ||
    lastName === '' ||
    phoneNumber.length < 8 ||
    password === '' ||
    cPassword === ''
  ) {
    Alert.alert('Fail', 'Fix errors in the form and try again!');
    console.log(firstName, lastName, phoneNumber, eMail, password, cPassword);
    return;
  }

  try {
    const response = await client.mutate({
      mutation: SIGN_UP_USER,
      variables: {
        username: phoneNumber,
        phone: phoneNumber,
        name: `${firstName} ${lastName}`,
        email: eMail,
        password: password,
      },
    });

    const {data} = response;
    const {signUp} = data;
    const {result} = signUp;

    setIsLoading(false);

    if (result === 'Success') {
      setRegistered(true);
    } else if (result === 'Failure') {
      Alert.alert(
        'Sign up failure.',
        'Check Phone number and E-mail. Press Ok to try again.',
      );
    } else {
      Alert.alert('Oops!', 'Try again!');
    }
  } catch (error) {
    console.error(error);
    Alert.alert(
      'Failure',
      'Something wrong happened. Check your internet and try again!',
    );
  }
};

export const signOut = async callback => {
  callback();
};

export const clearStorage = async () => {
  await AsyncStorage.clear();
};

export const autoLogin = async () => {
  let accessToken = null;

  try {
    let tokenString = await AsyncStorage.getItem('tokens');

    let tokens = tokenString !== null && JSON.parse(tokenString);
    accessToken = tokens.accessToken;
  } catch (e) {
    // Restoring token failed
    console.log(e);
    console.log('Restoring token failed');
    console.log('acessToken ', accessToken);
  }
};

export const getKeys = async () => {
  const keys = await AsyncStorage.getAllKeys();
  console.log(keys);
};

const processMessage = async () => {};
