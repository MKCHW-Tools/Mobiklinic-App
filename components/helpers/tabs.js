/* import React from 'react'
import {createMaterialTopTabNavigator } from 'react-navigation-tabs'
import Check from '../views/checker'
import Stats from '../views/stats'
import Info from '../views/info'

const Tabs = createMaterialTopTabNavigator({ Check, Stats, Info },
    {
        tabBarOptions: {
            labelStyle: {
                fontSize: 12,
            },
            style: {
                backgroundColor: '#C1282D',
            },
            indicatorStyle:{
                backgroundColor: 'rgba(255,255,255, 0.29)'
            }
        },
        navigationOptions:{
            headerStyle: {
                backgroundColor: '#C1282D'
            },
            headerTintColor: '#fff',
        }
    }
)

export default Tabs */