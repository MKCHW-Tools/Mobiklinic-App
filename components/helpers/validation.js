
    export const validateAlphaNum = input => {
        let pattern = /([a-zA-Z])\w+/ig
        return pattern.test(input)
    }

    export const validateNumber = input => {
        let pattern = /(^[\d])\w+/g
        return pattern.test(input)
    }

    export const validateUgandaPhoneNumber = input => {
        let pattern = /(?:(256))(3|4|7)\d{2}\d{6}/
        return pattern.test(input)
    }

    export const validateEmail = input => {
        let pattern = /^\S+@\S+[\.][0-9a-z]+$/
        return pattern.test(input)
    }

    export const validatePassword = input => {
        let pattern = /^(?=.{8,32})(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=]).*$/
        return pattern.test(input)
    }

    export const similarPasswords = (input1, input2) => input1 === input2
