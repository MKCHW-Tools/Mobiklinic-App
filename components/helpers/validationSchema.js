import * as Yup from 'yup';
import React from 'react';

const validationSchema = Yup.object({
  personalInformation: Yup.object().shape({
    fullname: Yup.string().required('Full Name is required'),
    age_group: Yup.string().required('Age Group is required'),
    phone: Yup.string().required('Phone number is required'),
    condition: Yup.string().required('Condition is required'),
  }),
  medicalConditions: Yup.object().shape({
    currentConditions: Yup.string(),
    pastConditions: Yup.string(),
    chronicDiseases: Yup.string(),
    allergies: Yup.string(),
    surgeries: Yup.string(),
  }),
  medications: Yup.object().shape({
    currentMedications: Yup.string(),
    previousMedications: Yup.string(),
    medicationAllergies: Yup.string(),
  }),
  familyMedicalHistory: Yup.object().shape({
    familyDiseases: Yup.string(),
    hereditaryConditions: Yup.string(),
  }),
  reproductiveHistory: Yup.object().shape({
    pregnancyHistory: Yup.string(),
    obstetricGynecologicalHistory: Yup.string(),
  }),
  additionalInformation: Yup.object().shape({
    otherMedicalInformation: Yup.string(),
  }),
});
export default validationSchema;
