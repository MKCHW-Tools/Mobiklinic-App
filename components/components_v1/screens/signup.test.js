
import 'react-native';
import React from 'react';
import { Alert } from 'react-native';
import SignUp from "../../screens/signup";
import { render, fireEvent } from '@testing-library/react-native';
import * as funcs from '../../helpers/functions';

import renderer from 'react-test-renderer';


describe("Testing Sign Up Component", () => {

    //mock the navigation
    const mockNavigation = {
        navigate: jest.fn()
    };

    //Tests that the component renders correctly
    it('renders same screen', () => {
        const tree = renderer.create(
            <SignUp />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('renders field lables and placeholders', () => {
        const { getByText, getByPlaceholderText } = render(<SignUp navigation={mockNavigation} />);

        // Verify the existence of UI elements like labels, placeholders, and buttons
        expect(getByText('Sign Up')).toBeTruthy();
        expect(getByPlaceholderText('Enter First name')).toBeTruthy();
        expect(getByPlaceholderText('Enter Last name')).toBeTruthy();

    });


    // Tests that the user is registered when valid data is submitted during sign up
    it('should register user when valid data is submitted', async () => {
        const data = {
            firstName: 'Silas',
            lastName: 'sangmin',
            phoneNumber: '256754123456',
            password: 'password',
            cPassword: 'password',
            eMail: 'Silassangmin@example.com',
            setIsLoading: jest.fn(),
            setProcess: jest.fn(),
            setRegistered: jest.fn()
        };

        //const signUpMock = jest.spyOn(funcs, 'signUp').mockImplementation(() => Promise.resolve());
        const signUpMock = jest.spyOn(funcs, 'signUp');

        const alertSpy = jest.spyOn(Alert, 'alert').mockImplementation((title, message, buttons) => {
            buttons[0].onPress();
        });

        const { getByText, getByPlaceholderText } = render(<SignUp />);
        const signUpButton = getByText('Sign Up');
        fireEvent.press(signUpButton);
        expect(signUpMock).toBeCalled();
        //expect(alertSpy).toBeCalled();
        expect(alertSpy).toHaveBeenCalledWith('Error', 'Please fill in the required fields correctly !', [
            {
                text: 'OK',
                onPress: expect.any(Function)
            }
        ]); //should alert error because default fields are empty
    });

    //Test Already have an account? Sign in navigation to sign in screen
    it('should navigate to sign in screen', async () => {
        const { getByText } = render(<SignUp navigation={mockNavigation} />);
        const signInNavButton = getByText('Already have an account? Sign in');
        fireEvent.press(signInNavButton);
        expect(mockNavigation.navigate).toHaveBeenCalledWith('Login');
    });
});

