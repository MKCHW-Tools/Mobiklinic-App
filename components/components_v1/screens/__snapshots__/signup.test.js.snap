// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Sign Up Component renders same screen 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 3,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexGrow": 2,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Image
      source=""
      style={
        {
          "height": 80,
          "width": 80,
        }
      }
    />
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 14,
          "fontWeight": "bold",
          "padding": 10,
          "textTransform": "uppercase",
        }
      }
    >
      Sign up to become A Mobiklinic CHP.
    </Text>
  </View>
  <RCTScrollView
    style={
      {
        "paddingTop": 50,
      }
    }
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "#42a5f5",
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "flexGrow": 1,
            "justifyContent": "center",
            "padding": 20,
          }
        }
      >
        <View>
          <Text
            style={
              {
                "color": "#f00",
              }
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(255,255,255,0.6)",
              "borderColor": "#ccc",
              "borderRadius": 18,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000",
              "flexDirection": "row",
              "fontFamily": "Roboto",
              "fontSize": 15,
              "marginBottom": 15,
              "marginHorizontal": 15,
              "paddingHorizontal": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            First Name:
          </Text>
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Enter First name"
            placeholderTextColor="grey"
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 15,
                "fontWeight": "medium",
                "justifyContent": "center",
                "paddingLeft": 10,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(255,255,255,0.6)",
              "borderColor": "#ccc",
              "borderRadius": 18,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000",
              "flexDirection": "row",
              "fontFamily": "Roboto",
              "fontSize": 15,
              "marginBottom": 15,
              "marginHorizontal": 15,
              "paddingHorizontal": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Last Name:
          </Text>
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Enter Last name"
            placeholderTextColor="grey"
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 15,
                "fontWeight": "medium",
                "justifyContent": "center",
                "paddingLeft": 10,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(255,255,255,0.6)",
              "borderColor": "#ccc",
              "borderRadius": 18,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000",
              "flexDirection": "row",
              "fontFamily": "Roboto",
              "fontSize": 15,
              "marginBottom": 15,
              "marginHorizontal": 15,
              "paddingHorizontal": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Email Address:
          </Text>
          <TextInput
            autoCorrect={false}
            keyboardType="email-address"
            onChangeText={[Function]}
            placeholder="Enter Email"
            placeholderTextColor="grey"
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 15,
                "fontWeight": "medium",
                "justifyContent": "center",
                "paddingLeft": 10,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(255,255,255,0.6)",
              "borderColor": "#ccc",
              "borderRadius": 18,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000",
              "flexDirection": "row",
              "fontFamily": "Roboto",
              "fontSize": 15,
              "marginBottom": 15,
              "marginHorizontal": 15,
              "paddingHorizontal": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Phone Number:
          </Text>
          <TextInput
            autoCorrect={false}
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="eg '256754XXXXXXX'"
            placeholderTextColor="grey"
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 15,
                "fontWeight": "medium",
                "justifyContent": "center",
                "paddingLeft": 10,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(255,255,255,0.6)",
              "borderColor": "#ccc",
              "borderRadius": 18,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000",
              "flexDirection": "row",
              "fontFamily": "Roboto",
              "fontSize": 15,
              "marginBottom": 15,
              "marginHorizontal": 15,
              "paddingHorizontal": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Password:
          </Text>
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            password={true}
            placeholder="Enter Password"
            placeholderTextColor="grey"
            secureTextEntry={true}
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 15,
                "fontWeight": "medium",
                "justifyContent": "center",
                "paddingLeft": 10,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(255,255,255,0.6)",
              "borderColor": "#ccc",
              "borderRadius": 18,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000",
              "flexDirection": "row",
              "fontFamily": "Roboto",
              "fontSize": 15,
              "marginBottom": 15,
              "marginHorizontal": 15,
              "paddingHorizontal": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Confirm :
          </Text>
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            password={true}
            placeholder="Enter Password again"
            placeholderTextColor="grey"
            secureTextEntry={true}
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 15,
                "fontWeight": "medium",
                "justifyContent": "center",
                "paddingLeft": 10,
              }
            }
            value=""
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#7cd2ff",
              "borderRadius": 10,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 15,
              "marginVertical": 10,
              "opacity": 1,
              "padding": 10,
              "paddingHorizontal": 15,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "fontWeight": "bold",
                "paddingHorizontal": 15,
              }
            }
          >
            Sign Up
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            strokeSize={3}
            style={
              [
                {
                  "color": "#000",
                  "fontSize": 20,
                },
                undefined,
                {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              [
                {
                  "paddingTop": 10,
                  "textAlign": "center",
                  "textDecorationLine": "underline",
                },
              ]
            }
          >
            Already have an account? Sign in
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
